internal/cloudapi/v2/compose_test.go

```diff
@@ -848,6 +848,33 @@ func TestGetImageRequests_BlueprintDistro(t *testing.T) {
 	assert.Equal(t, got[0].blueprint.Distro, "fedora-39")
 }
 
+// TestGetImageRequests_BlueprintArch test to make sure blueprint architecture overrides
+// the request arch
+func TestGetImageRequests_BlueprintArch(t *testing.T) {
+	uo := UploadOptions(struct{}{})
+	request := &ComposeRequest{
+		Distribution: "fedora-40",
+		ImageRequest: &ImageRequest{
+			Architecture:  "x86_64",
+			ImageType:     ImageTypesAws,
+			UploadOptions: &uo,
+			Repositories:  []Repository{},
+		},
+		Blueprint: &Blueprint{
+			Name:         "arch-test",
+			Architecture: common.ToPtr("aarch64"),
+		},
+	}
```

Comment:
IMHO, this compose request should not be treated as valid. The architecture values don't match.